From 1e0b46a476c1c950e22403a2d8950f5c97a1a0fe Mon Sep 17 00:00:00 2001
From: Martin Jansa <martin.jansa@lge.com>
Date: Fri, 5 Jun 2020 11:35:25 -0700
Subject: [PATCH] CMakeLists.txt: prevent building zstd with ExternalProject

* use pkg-config, because zstd doesn't provide cmake find

Upstream-Status: Pending

Signed-off-by: Martin Jansa <martin.jansa@lge.com>

Rebase patch on iron release.

Signed-off-by: Sandeep Gundlupet Raju <sandeep.gundlupet-raju@amd.com>
---
 CMakeLists.txt                      | 54 +++--------------------------
 cmake_minimum_required_2.8.12.patch | 26 --------------
 no_internal_headers.patch           | 27 ---------------
 3 files changed, 4 insertions(+), 103 deletions(-)
 delete mode 100644 cmake_minimum_required_2.8.12.patch
 delete mode 100644 no_internal_headers.patch

Index: git/CMakeLists.txt
===================================================================
--- git.orig/CMakeLists.txt
+++ git/CMakeLists.txt
@@ -12,52 +12,12 @@ option(FORCE_BUILD_VENDOR_PKG
 
 if(NOT FORCE_BUILD_VENDOR_PKG)
   # We need at least VERSION 1.4.8, version check is done in Findzstd.cmake
-  find_package(zstd 1.4.8 QUIET)
+  find_package(PkgConfig REQUIRED)
+  pkg_check_modules(ZSTD REQUIRED libzstd>=1.4.8)
 endif()
 
-macro(build_zstd)
-  set(extra_cmake_args)
-
-  if(DEFINED CMAKE_BUILD_TYPE)
-    list(APPEND extra_cmake_args -DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE})
-  endif()
-
-  if(DEFINED CMAKE_TOOLCHAIN_FILE)
-    list(APPEND extra_cmake_args "-DCMAKE_TOOLCHAIN_FILE=${CMAKE_TOOLCHAIN_FILE}")
-  else()
-    list(APPEND extra_cmake_args "-DCMAKE_CXX_COMPILER=${CMAKE_CXX_COMPILER}")
-  endif()
-  list(APPEND extra_cmake_args "-DCMAKE_CXX_FLAGS=${CMAKE_CXX_FLAGS}")
-
-  set(zstd_version 1.4.8)
-  include(ExternalProject)
-  # The CMakeLists.txt file for zstd is in a subdirectory.
-  # We need to configure the CMake command to build from there instead.
-  externalproject_add(zstd-${zstd_version}
-    GIT_REPOSITORY https://github.com/facebook/zstd.git
-    GIT_TAG 97a3da1df009d4dc67251de0c4b1c9d7fe286fc1 # v1.4.8
-    GIT_CONFIG advice.detachedHead=false
-    # Suppress git update due to https://gitlab.kitware.com/cmake/cmake/-/issues/16419
-    # See https://github.com/ament/uncrustify_vendor/pull/22 for details
-    UPDATE_COMMAND ""
-    TIMEOUT 60
-    SOURCE_SUBDIR build/cmake
-    CMAKE_ARGS
-      -DCMAKE_INSTALL_PREFIX=${CMAKE_CURRENT_BINARY_DIR}/${PROJECT_NAME}_install
-      -DZSTD_BUILD_STATIC=OFF
-      -DZSTD_BUILD_SHARED=ON
-      -DZSTD_BUILD_PROGRAMS=OFF
-      ${extra_cmake_args}
-  )
-
-  install(
-    DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/${PROJECT_NAME}_install/
-    DESTINATION ${CMAKE_INSTALL_PREFIX}
-    USE_SOURCE_PERMISSIONS)
-endmacro()
-
-if(NOT zstd_FOUND)
-  build_zstd()
+if(NOT ZSTD_FOUND)
+  message(STATUS "Zstd not found, missing dependency or version less than 1.4.8, found ${ZSTD_VERSION}")
 else()
   message(STATUS "Found Zstd, skipping build.")
 endif()
